### PR TITLE
BS - Docker

### PR DESCRIPTION
**Pull request by egordunaev**
Adding CI/CD for the project, total list of features:

- [x] Building python package and check linting (flake8)
- [ ] Publish build in Github Package Registry
- [x] Being able to Dev-in-Container
- [x] Being able to run docker-compose

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python set up and linting
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (Zimtir) [inline .github/workflows/python-package.yml:10]:
```suggestion
  push:
  pull_request:
    branches: [ master ]
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      matrix:
+        python-version: [3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
```
Comment (Zimtir) [inline .github/workflows/python-package.yml:30]:
```suggestion

    - name: Install dependencies
```

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    defaults:
+      run:
+        working-directory: ./development
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.9]
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (Zimtir) [inline .github/workflows/python-package.yml:25]:
```suggestion
    - uses: actions/checkout@v2

```

File: .github/workflows/python-package.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
```
Comment (Zimtir) [inline .github/workflows/python-package.yml:35]:
```suggestion

    - name: Lint with flake8
```

File: development/.gitignore [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# pytype static type analyzer
+.pytype/
+
+# Cython debug symbols
+cython_debug/
```
Comment (Zimtir) [inline development/.gitignore:140]:
```suggestion
cython_debug/

```

**Review by Zimtir — CHANGES_REQUESTED** [review]

**Comment by Zimtir** [comment]
@egordunaev should we do something extra here before merge? I think we should move it forward

Does it make sense?

**Comment by egordunaev** [comment]
> @egordunaev should we do something extra here before merge? I think we should move it forward
> 
> Does it make sense?

@Zimtir 
I wanted to finish this checklist before merge, or is this PR needs to be complete ASAP? (Sorry for the delay)

**Comment by Zimtir** [comment]
> > @egordunaev should we do something extra here before merge? I think we should move it forward
> > Does it make sense?
> 
> @Zimtir
> I wanted to finish this checklist before merge, or is this PR needs to be complete ASAP? (Sorry for the delay)

I think we can avoid finishing the checklist because it can so long, but the current code can be used now

**Comment by egordunaev** [comment]
> > > @egordunaev should we do something extra here before merge? I think we should move it forward
> > > Does it make sense?
> > 
> > 
> > @Zimtir
> > I wanted to finish this checklist before merge, or is this PR needs to be complete ASAP? (Sorry for the delay)
> 
> I think we can avoid finishing the checklist because it can so long, but the current code can be used now

Would it be better if we made another task for this?

**Comment by Zimtir** [comment]
> > > > @egordunaev should we do something extra here before merge? I think we should move it forward
> > > > Does it make sense?
> > > 
> > > 
> > > @Zimtir
> > > I wanted to finish this checklist before merge, or is this PR needs to be complete ASAP? (Sorry for the delay)
> > 
> > 
> > I think we can avoid finishing the checklist because it can so long, but the current code can be used now
> 
> Would it be better if we made another task for this?

Could you make issues here?
https://github.com/future-code-lab/botnet/issues

**Comment by egordunaev** [comment]
@Zimtir PTAL 🥺 🙏

**Review by Zimtir — APPROVED** [review]
LGTM with small suggestions

File: development/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+version: "3.9"
+services:
+  bot-server:
+    build: .
+    ports:
+      - "5000:5000"
+
```
Comment (Zimtir) [inline development/docker-compose.yml:7]:
```suggestion

```

**Review by Zimtir — APPROVED** [review]

**Comment by Zimtir** [comment]
🐰

**Pull request merged**